src/pdk/value.rs

```diff
@@ -0,0 +1,68 @@
+use abi_stable::{
+    std_types::{RBox, RCow, RHashMap, RVec},
+    StableAbi,
+};
+use value_trait::StaticNode;
+
+/// Representation of a JSON object
+pub type Object<'value> = RHashMap<RCow<'value, str>, Value<'value>>;
+/// Bytes
+pub type Bytes<'value> = RCow<'value, [u8]>;
+
+/// FFI-safe `Value` type to communicate with the plugins. It's meant to be
+/// converted to/from the original `tremor_value::Value` type and back so that
+/// it can be passed through the plugin interface.
+#[repr(C)]
+#[derive(StableAbi)]
+pub enum Value<'value> {
```

Comment:
If this is the `Value` PDK developers get into their hands i think we need to implement more traits from `value_trait` on this one, to make it usable. But this is not a concern for this PR. Please add a Note about it at least.

Reply:
Supposedly it's only used as an intermediate. The regular value is converted to the FFI one, and the in the plugin it's converted back to the regular one.

I still have to figure out how to do these conversions in the least annoying way for the plugin developer but it's definitely doable.

Reply:
Got it, i wasn't aware of how this was handled on the plugin side. 👍